fabric-data-generation-api-v1/src/client/java/net/fabricmc/fabric/impl/datagen/client/SoundTypeBuilderImpl.java

```diff
@@ -0,0 +1,217 @@
+
+	private SoundCategory category = SoundCategory.NEUTRAL;
+	@Nullable
+	private String subtitle;
+	private final List<Entry> sounds = new ArrayList<>();
+
+	public SoundTypeBuilderImpl() { }
+
+	@Override
+	public SoundTypeBuilder category(SoundCategory category) {
+		Objects.requireNonNull(category, "Sound event category must not be null.");
+		this.category = category;
+		return this;
+	}
+
+	@Override
+	public SoundTypeBuilder subtitle(@Nullable String subtitle) {
+		this.subtitle = subtitle;
+		return this;
+	}
+
+	@Override
+	public SoundTypeBuilder sound(EntryBuilder sound) {
+		Objects.requireNonNull(sound, "Sound must not be null.");
+		sounds.add(((EntryBuilderImpl) sound).build(""));
+		return this;
+	}
+
+	@Override
+	public SoundTypeBuilder sound(EntryBuilder sound, int count) {
+		Objects.requireNonNull(sound, "Sound must not be null.");
+		Preconditions.checkArgument(count > 0, "Count must be greater than zero.");
+
+		for (int i = 1; i <= count; i++) {
+			sounds.add(((EntryBuilderImpl) sound).build(Integer.toString(i)));
+		}
+
+		return this;
+	}
+
+	public SoundType build() {
+		Preconditions.checkState(!sounds.isEmpty(), "Sound definition must have at least one sound file");
+
+		for (Entry sound : sounds) {
+			if (sound.type() == RegistrationType.SOUND_EVENT) {
+				Registries.SOUND_EVENT.getOptionalValue(sound.name()).orElseThrow(() -> new IllegalStateException("Referenced sound event " + sound.name() + " does not exist"));
+			}
+		}
+
+		return new SoundType(sounds, category, Optional.ofNullable(subtitle));
+	}
+
+	public record SoundType(List<Entry> sounds, SoundCategory category, Optional<String> subtitle) {
+		private static final Map<String, SoundCategory> CATEGORIES = Arrays.stream(SoundCategory.values()).collect(Collectors.toMap(SoundCategory::getName, Function.identity()));
+		private static final Codec<SoundCategory> SOUND_CATEGORY_CODEC = Codec.stringResolver(SoundCategory::getName, name -> CATEGORIES.getOrDefault(name.toLowerCase(Locale.ROOT), SoundCategory.NEUTRAL));
+		public static final Codec<SoundType> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+				Entry.CODEC.listOf().fieldOf("sounds").forGetter(SoundType::sounds),
+				SOUND_CATEGORY_CODEC.fieldOf("category").forGetter(SoundType::category),
+				Codec.STRING.optionalFieldOf("subtitle").forGetter(SoundType::subtitle)
+		).apply(instance, SoundType::new));
+	}
+
+	private record Entry(Identifier name, RegistrationType type, float volume, float pitch, int weight, int attenuationDistance, boolean stream, boolean preload) {
+		private static final Codec<Entry> MAP_CODEC = RecordCodecBuilder.create(instance -> instance.group(
+				Identifier.CODEC.fieldOf("name").forGetter(Entry::name),
+				RegistrationType.CODEC.optionalFieldOf("type", RegistrationType.FILE).forGetter(Entry::type),
+				Codec.FLOAT.optionalFieldOf("volume", 1F).forGetter(Entry::volume),
```

Comment:
Nit: Use Codecs.POSITIVE_FLOAT and Codecs.POSITIVE_INT to match the logic in SoundEntryDeserializer.

Edit, just seen you handle this in the builder, so disregard this.